File: Tarefa5/main.py
from utils import integracao


if __name__ == "__main__":

    print("---------------------------------------------")
    print("-------------------Tarefa 5------------------")
    print("---------------------------------------------\n")

    programa = True

    while programa:
        print("A funcao usada sera: (sen(2x) + 4*(x)^2 + 3x)^2\n")
        xi = int(input("Escreva o valor de xi:"))
        xf = int(input("Escreva o valor de xf:"))
        epilson = float(input("Escreva o valor do erro:"))

        print("Escolha quantos pontos a Quadratura de Gauss-Legendre: ")
        print("0: 2 pontos ")
        print("1: 3 pontos ")
        print("2: 4 pontos ")
        pontos = int(input("Digite qual tipo de Quadratura de Gauss-Legendre voce quer usar: "))
        
        if pontos == 0:
            print("Você escolheu  Quadratura de Gauss-Legendre de 2 pontos")
            integracao(xi, xf, epilson, pontos)
        if pontos == 1:
            print("Você escolheu  Quadratura de Gauss-Legendre de 3 pontos")
            integracao(xi, xf, epilson, pontos)
        if pontos == 2:
            print("Você escolheu  Quadratura de Gauss-Legendre de 4 pontos")
            integracao(xi, xf, epilson, pontos)      
            
        print("\n---------------------------------------------")
        print("1-Rodar o programa novamente ")
        print("2-Sair do programa")
        print("---------------------------------------------")

        novamente = int(input("Digite a sua escolha?"))

        if novamente == 1:
            programa = True
        if novamente == 2:
            programa = False

File: Tarefa5/main.cpp
#include <iostream>
#include <cmath>
#include "utils.cpp"

using namespace std;

int main() {

    cout << "---------------------------------------------" << endl;
    cout << "-------------------Tarefa 5------------------" << endl;
    cout << "---------------------------------------------\n" << endl;

    bool programa = true;

    while (programa) {
        cout << "A funcao usada sera: (sen(2x) + 4*(x)^2 + 3x)^2\n" << endl;
        int xi;
        cout << "Escreva o valor de xi:";
        cin >> xi;
        int xf;
        cout << "Escreva o valor de xf:";
        cin >> xf;
        double epilson;
        cout << "Escreva o valor do erro:";
        cin >> epilson;

        cout << "Escolha quantos pontos a Quadratura de Gauss-Legendre: " << endl;
        cout << "0: 2 pontos " << endl;
        cout << "1: 3 pontos " << endl;
        cout << "2: 4 pontos " << endl;
        int pontos;
        cout << "Digite qual tipo de Quadratura de Gauss-Legendre voce quer usar: ";
        cin >> pontos;
        
        if (pontos == 0) {
            cout << "Você escolheu  Quadratura de Gauss-Legendre de 2 pontos" << endl;
            integracao(xi, xf, epilson, pontos);
        } else if (pontos == 1) {
            cout << "Você escolheu  Quadratura de Gauss-Legendre de 3 pontos" << endl;
            integracao(xi, xf, epilson, pontos);
        } else if (pontos == 2) {
            cout << "Você escolheu  Quadratura de Gauss-Legendre de 4 pontos" << endl;
            integracao(xi, xf, epilson, pontos);
        }      
            
        cout << "\n---------------------------------------------" << endl;
        cout << "1-Rodar o programa novamente " << endl;
        cout << "2-Sair do programa" << endl;
        cout << "---------------------------------------------" << endl;

        int novamente;
        cout << "Digite a sua escolha?";
        cin >> novamente;

        if (novamente == 1) {
            programa = true;
        } else if (novamente == 2) {
            programa = false;
        }
    }

    return 0;
}
